[nibeheatpump] Readme for NibeGW for Arduino with Hardware and compilation hints (#13787)
openhab/openhab-addons: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md

# NibeGW Hardware and Compiling hints

## RS-485 Modules

> For setttings up a NibeGW you need a RS485 module. Most cheap modules out there are compatible with 5V voltage and therefore compatible with Arduino based hardware. Mostly you will get one of two commonly used designs:

Reviewer: ProDiNo boards have RS-485 port build in, so would it make sense to say that if you board doesn't have rs-485 port you need a rs-485 module.

Author: Good point - added.

* Modules based on Max1348 chip, which has 2 pins on the Arduino side (RXD, TXD) + VCC + GND
* Modules based on Max485 chip, which has 4 pins on the Arduino side (RO, RE, DE, DI) + VCC + GND

Both types of modules work fine with NibeGW. The difference between the two is, that the Max485 chip needs to be switched between RX mode and TX mode manually while the Max1348 chip do this automatically. That is why you need an extra "direction pin" on the Arduino to switch the module with Max485 chip between the two modes.

#### Wiring diagram for Max1348 based modules:

```
 TX  RX   5V  GND    Arduino
 |   |    |    |
 |   |    |    |
 |   |    |    |
 RX  TX  VCC  GND    Max1348 based module
```

#### Wiring diagram for Max485 (PIN2 is used as direction pin here):

```
 TX  RX  PIN2      5V  GND    Arduino
 |   |    |        |    |
 |   |    |        |    |
 |   |    |---|    |    |
 DI  RO   DE  RE  VCC  GND    Max485 bases module
```

> ## Ethernet Shield W5100

Reviewer: ProDiNo boards have also Ethernet build in, so separated shield is not needed

Author: Good point - added.

* Modules based on Max1348 chip, which has 2 pins on the Arduino side (RXD, TXD) + VCC + GND
* Modules based on Max485 chip, which has 4 pins on the Arduino side (RO, RE, DE, DI) + VCC + GND

Both types of modules work fine with NibeGW. The difference between the two is, that the Max485 chip needs to be switched between RX mode and TX mode manually while the Max1348 chip do this automatically. That is why you need an extra "direction pin" on the Arduino to switch the module with Max485 chip between the two modes.

#### Wiring diagram for Max1348 based modules:

```
 TX  RX   5V  GND    Arduino
 |   |    |    |
 |   |    |    |
 |   |    |    |
 RX  TX  VCC  GND    Max1348 based module
```

#### Wiring diagram for Max485 (PIN2 is used as direction pin here):

```
 TX  RX  PIN2      5V  GND    Arduino
 |   |    |        |    |
 |   |    |        |    |
 |   |    |---|    |    |
 DI  RO   DE  RE  VCC  GND    Max485 bases module
```

## Ethernet Shield W5100

> This Ethernet shield is based on Wiznet W5100 Ethernet Chip. It is supported by the Adruino Ehternet Library. No special configuration is needed, NibeGW supports this shield out of the box.

Reviewer 2: Typo or intentional?
```suggestion
This Ethernet shield is based on Wiznet W5100 Ethernet Chip. It is supported by the Arduino Ethernet Library. No special configuration is needed, NibeGW supports this shield out of the box.
```

ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield. Also the ProDiNo Ethernet is supported by NibeGW out of the box.

## Arduino Uno

> Arduino Uno has only one serial port which is shared with usb. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino. Furthermore do not use the usb port while Arduino is communicating with the Nibe heatpump.

Reviewer 2: ```suggestion
Arduino Uno has only one serial port which is shared with USB. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino. Furthermore do not use the USB port while Arduino is communicating with the Nibe heatpump.
```

# NibeGW Hardware and Compiling hints

## RS-485 Modules

For settting up a NibeGW you need a RS485 module. 

> While ProDiNo already have RS-485 support included, you need a seperate module for Arduino Uno.

Reviewer 2: ```suggestion
While ProDiNo already have RS-485 support included, you need a separate module for Arduino Uno.
```

* Modules based on Max1348 chip, which has 2 pins on the Arduino side (RXD, TXD) + VCC + GND
* Modules based on Max485 chip, which has 4 pins on the Arduino side (RO, RE, DE, DI) + VCC + GND

Both types of modules work fine with NibeGW.
The difference between the two is, that the Max485 chip needs to be switched between RX mode and TX mode manually while the Max1348 chip do this automatically.
That is why you need an extra "direction pin" on the Arduino to switch the module with Max485 chip between the two modes.

#### Wiring diagram for Max1348 based modules:

```
 TX  RX   5V  GND    Arduino
 |   |    |    |
 |   |    |    |
 |   |    |    |
 RX  TX  VCC  GND    Max1348 based module
```

#### Wiring diagram for Max485 (PIN2 is used as direction pin here):

```
 TX  RX  PIN2      5V  GND    Arduino
 |   |    |        |    |
 |   |    |        |    |
 |   |    |---|    |    |
 DI  RO   DE  RE  VCC  GND    Max485 bases module
```

## Ethernet Shield W5100

This Ethernet shield is based on Wiznet W5100 Ethernet Chip.
It is supported by the Arduino Ehternet Library.
No special configuration is needed, NibeGW supports this shield out of the box.

> ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield.

Reviewer 2: ```suggestion
ProDiNo already have Ethernet included, so there's no need for a separate Ethernet Shield.
```

## Ethernet Shield W5100

This Ethernet shield is based on Wiznet W5100 Ethernet Chip.
It is supported by the Arduino Ehternet Library.
No special configuration is needed, NibeGW supports this shield out of the box.

ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield.
Also the ProDiNo Ethernet is supported by NibeGW out of the box.

## Arduino Uno

> Arduino Uno has only one serial port which is shared with USB. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino.

Reviewer 2: Can you divide this line also? :)

Author: Done.
